### PR TITLE
Add Meier function

### PR DESCRIPTION
**Pull request by hankwustfc**
Description of work:

Add Meier to Mantid fit functions.

To test:

1.Test if the documentation is added
2.Test if the fit function is working
3.Test the function with the unit test file
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
will be added to release notes as part of #26827

**Comment by hankwustfc** [comment]
Screenshot 2019-09-23 at 10 37 46

For your reference

File: docs/source/fitting/fitfunctions/Meier.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+.. math:: P_z(t) = \frac{1}{2J+1}\left\{1+\sum^J_{m=-J+1}[\cos^2(2\alpha_m)+\sin^2(2\alpha_m)\cos(\lambda^+_m-\lambda^-_m)t]\right\},
+
+.. math:: P_x(t) = \frac{1}{2J+1}\sum^J_{m=-J} \{ \cos^2\alpha_{m+1}\sin^2\alpha_m\cos(\lambda_{m+1}^+-\lambda_m^+)t +\cos^2\alpha_{m+1}\cos^2\alpha_m\cos(\lambda_{m+1}^+-\lambda_m^-)t +\sin^2\alpha_{m+1}\sin^2\alpha_m\cos(\lambda_{m+1}^--\lambda_m^+)t +\sin^2\alpha_{m+1}\cos^2\alpha_m\cos(\lambda_{m+1}^--\lambda_m^-)t\},
+	
+..math:: \lambda_m^\pm = \frac{1}{2}[\omega_Q(2m^2-2m+1)+\omega_D\pm W_m],
+
+..math:: W_m = \{(\omega_D+\omega_Q)^2(2m-1)^2+\omega_D^2[J(J+1)-m(m-1)]\}^\frac{1}{2},
+
+..math:: tan(2\alpha_m)=\frac{\omega_D[J(J+1)-m(m-1)]^\frac{1}{2}}{(1-2m)(\omega_D+\omega_Q)},
```
Comment (NickDraper) [inline docs/source/fitting/fitfunctions/Meier.rst]: You need the space between the .. and the math.

so .. math
not ..math

I'll fix it myself this time

**Review by NickDraper — DISMISSED** [review]
Now this is good to go, docs in place, tests and code is ok

**Review by NickDraper — DISMISSED** [review]
tests, docs and code ok

**Review by NickDraper — APPROVED** [review]
tests, docs and code ok

**Pull request merged**